### ISSUE TITLE
"Run unsafe queries without a WHERE clause" when set escaped values on update string

### ISSUE THREAD

**Issue by Newcomer1989** [HeidiSQL/HeidiSQL#1769]
**Preconditions**

* HeidiSQL version: 12.0.0.6468
* Database type and version: PostgreSQL 14
* OS: Windows 10 Pro

**Describe the bug**

When using escape patterns inside the update-value, HeidiSQL sends a wrong dialog warning:

> Your query contrains UPDATEs and/or DELETEs without a WHERE clause. Please confirm that you know what you're doing.

In this case, it isn't an SQL statement without a WHERE condition. The sent SQL statement did its job right and updates only a few rows, not the whole table.

**To Reproduce**

Run an update command like:
`UPDATE table SET column_1='&''### HIN-Satz''' WHERE column_2='$''### HIN-Satz''';`

**Screenshots or Crash reports**
![image](https://user-images.githubusercontent.com/11364569/220926408-0e13e396-440f-40eb-aded-f21837ffc5c9.png)

**Comment by Newcomer1989** [comment]
This error is reproducible with version 12.3.0.6589 (portable version).

**Comment by ansgarbecker** [comment]
Yes, can confirm that. String escaping is horribly hardcoded in HeidiSQL. Due to its MySQL focus mainly using backslash in most places. Will be quite tricky to fix, while the benefit is small.

**Comment by ansgarbecker** [comment]
Related: https://stackoverflow.com/q/74490745/4110077

**Comment by JP95Git** [comment]
I also got this message with this query:
UPDATE `Station` SET `WigosLocalIdentifier`=NULL WHERE `StationID`=36;

HeidiSQL 12.5.0.6677 on Windows 11 Pro while connected to a MariaDB database.

**Comment by ansgarbecker** [comment]
Still reproducible with v12.16.
